src/app/CommandHandler.h

```diff
@@ -77,6 +83,27 @@ class CommandHandler : public Messaging::ExchangeDelegate
         virtual Protocols::InteractionModel::Status CommandExists(const ConcreteCommandPath & aCommandPath) = 0;
     };
 
+    /**
+     * Class that allows asynchronous command processing before sending a
+     * response.  When such processing is desired:
+     *
+     * 1) Create a Handle initialized with the CommandHandler that delivered the
+     *    incoming command.
+     * 2) Ensure the Handle, or some Handle it's moved into via the move
+     *    constructor or move assignment operator, remains alive during the
+     *    course of the asynchronous processing.
+     * 3) Ensure that the ConcreteCommandPath involved will be known when
+     *    sending the response.
+     * 4) When ready to send the response:
```

Comment:
It would be good to add some text around ensuring the caller is managing the external synchronization requirements (i.e call with the lock held, or on the right threading context).

Reply:
To that end, do you think we should add a call to `assertChipStackLockedByCurrentThread` in `Handle::Get` below. As we saw yesterday, the nature of async work lends itself to easily violating the synchronization requirements, so adding that check will save most people a lot of headaches.

Reply:
That is a very good idea.  Will do.